components/omega/doc/design/VertCoord.md

```diff
@@ -0,0 +1,264 @@
+In the future, tidal potential and self attraction and loading will also have optional (default off) contributions to the geopotential.
+
+
+### 2.5 Requirement: Compute and store the desired thickness used in the $p^\star$ vertical coordinate
+
+The $p^\star$ vertical coordinate specifies a desired layer thickness that stretches the initial thicknesses by the bottom pressure anomaly.
+This involves adding a vertically-weighted bottom pressure perturbation to the reference (initial) thickness.
+This desired thickness will be computed and stored within the vertical coordinate module.
+The vertical advection algorithm will use this desired thickness to compute the vertical transport.
+
+### 2.6 Desired: General Arbitrary Lagrangian Eulerian (ALE) coordinate support
+
+In future versions of Omega, the $p^\star$ coordinate will be extended to a more general ALE coordinate
+
+### 2.7 Desired: Vertical Lagrangian remapping (VLR) support
+
+The vertical coordinate module should retain flexibility to support VLR in future versions of Omega.
+
+## 3 Algorithmic Formulation
+In the layered non-Boussinesq equations solved in Omega (see [V0 governing equation document](OmegaV1GoverningEqns) for details), the prognostic variable for cell $i$ and layer $k$ is the pseudo thickness, $\tilde{h}_{i,k}$, so that the geometric thickness (in meters) is a diagnostic variable defined as:
+
+$$ \Delta z_{i,k} = \rho_0 \alpha_{i,k} \tilde{h}_{i,k}. $$
+
+The pressure at vertical cell interfaces is the found by summing the pseudo thicknesses:
+
+$$  p_{i,k+1/2} = p_{i}^{surf} + g\rho_0 \sum_{k^\prime=1}^k \tilde{h}_{i,k^\prime}. $$
+
+and at the midpoint by
+
+$$ p_{i,k} = p_{i}^{surf} + g\rho_0 \sum_{k'=1}^{k-1} \tilde{h}_{i,k'} + \frac{1}{2} g\rho_0 \tilde{h}_{i,k} $$
+
+The $z$ location of cell interfaces is found by summing the pseudo thicknesses from the bottom multiplied by the specific volume:
+
+$$ z_{i,k}^{top} = z_i^{floor} + \rho_0 \sum_{k^\prime=k}^{K_{max}} \alpha_{i,k^\prime} \tilde{h}_{i,k^\prime}, $$
+
+where $z_i^{floor}$ is the (positive-up) bottom elevation.
+The $z$ location of cell centers is given by: 
+
+$$ z_{i,k} = z_i^{floor} + \frac{1}{2} \rho_0\alpha_{i,k} \tilde{h}_{i,k} + \rho_0\sum_{k^\prime= k+1}^{K_{max}} \alpha_{i,k^\prime} \tilde{h}_{i,k^\prime}. $$
+
+Initially, the geopotential is the sum of the $z$ height times $g$. 
+In the future, it will include contributions from the tidal potential ($\Phi_{tp}$) and self attraction and loading ($\Phi_{SAL}$):
+
+$$ \Phi_{i,k} = \left( gz_{i,k} + \Phi_{tp} + \Phi_{SAL} \right). $$
+
+The desired pseudo thickness used for the $p^\star$ coordinate is:
+
+$$\tilde{h}_k^{p^\star} = \tilde{h}_k^{rest} + \tilde{h}_k^{p}, $$
+
+where the pressure alteration, $\tilde{h}_k^{p}$, is given by:
+
+$$ \tilde{h}_k^{p} =\frac{(p_B-p_{surf})}{g\rho_0} \frac{W_k\tilde{h}_k^{rest}}{\sum_{k^\prime=1}^K W_{k^\prime}\tilde{h}_{k^\prime}^{rest}}.$$
```

Comment:
The weights $W_k$ should be explained.

Reply:
I don't think this equation and the previous one are consistent.  This seems like the full pseudo-thickness that each layer should have, not just the perturbation from the resting thickness.  I would benefit from a discussion of this if I'm missing something.

Reply:
I see. Sure, that sounds good.